funcr/example_test.go

```diff
@@ -34,3 +34,77 @@ func ExampleUnderlier() {
 	}
 	// Output: hello world
 }
+
+// newStdoutLogger returns a logr.Logger that prints to stdout.
+// It demonstrates how to implement a custom With* method which
+// controls another prefix for the output.
+func newStdoutLogger() logr.Logger {
+	l := &stdoutlogger{
+		Formatter: funcr.NewFormatter(funcr.Options{}),
+		prefix:    "default prefix",
+	}
+	return logr.New(l)
+}
+
+type stdoutlogger struct {
+	funcr.Formatter
+	prefix string
+}
+
+func (l stdoutlogger) WithName(name string) logr.LogSink {
+	l.Formatter.AddName(name)
+	return &l
+}
+
+func (l stdoutlogger) WithValues(kvList ...interface{}) logr.LogSink {
+	l.Formatter.AddValues(kvList)
+	return &l
+}
+
+func (l stdoutlogger) WithCallDepth(depth int) logr.LogSink {
+	l.Formatter.AddCallDepth(depth)
+	return &l
+}
+
+func (l stdoutlogger) Info(level int, msg string, kvList ...interface{}) {
+	prefix, args := l.FormatInfo(level, msg, kvList)
+	l.write(prefix, args)
+}
+
+func (l stdoutlogger) Error(err error, msg string, kvList ...interface{}) {
+	prefix, args := l.FormatError(err, msg, kvList)
+	l.write(prefix, args)
+}
+
+func (l stdoutlogger) write(prefix, args string) {
+	if prefix != "" {
+		fmt.Printf("%s: %s: %s\n", l.prefix, prefix, args)
+	} else {
+		fmt.Printf("%s: %s\n", l.prefix, args)
+	}
+}
+
+// withPrefix returns a copy of the logger with new settings for
+// the custom prefix. It returns the original logger if the underlying
+// LogSink is not a stdoutlogger.
+func withPrefix(log logr.Logger, prefix string) logr.Logger {
```

Comment:
Use a public name, since it's an example?

Reply:
Can we come up with something other than "prefix" which is confusing in `write()`.  E.g. change it to `FileLogger`, default to `os.Stdout`, and make this `WithFile(..., file *os.File)` ?

At that point it's still trivial but actually useful.

Reply:
and obviously write() would call `FPrint*()`

Reply:
> Use a public name, since it's an example?

I was wondering about that myself, too. Probably makes sense here and for the newStdoutLogger function. Will change.

> Can we come up with something other than "prefix" which is confusing in write().

True. I wanted something simple that is visible in the stdout. Replacing the output file would work only indirectly, for example because we could redirect to null and thus hide log messages.

What about a "log type" boolean? When true, messages are printed with INFO or ERROR in front of them depending on which function was used. When false, that's not done.

Reply:
I've pushed an update with that as additional feature.

Reply:
I don't follow "Replacing the output file would work only indirectly" - isn't the point here to make a plausible yet trivial demonstration?

Reply:
I see, for the purposes of a demo, changing the ouput FD isn't that helpful.  Fine.